src/main/java/de/triology/cas/logout/LogoutUriEnabledLogoutManagerImpl.java

```diff
@@ -0,0 +1,200 @@
+                final Service service = services.get(ticketId);
+                // it's a SingleLogoutService, else ignore
+                if (service instanceof SingleLogoutService) {
+                    final SingleLogoutService singleLogoutService = (SingleLogoutService) service;
+                    // the logout has not performed already
+                    if (!singleLogoutService.isLoggedOutAlready()) {
+                        final LogoutRequest logoutRequest = new LogoutRequest(ticketId, singleLogoutService);
+                        // always add the logout request
+                        logoutRequests.add(logoutRequest);
+                        final RegisteredService registeredService = servicesManager.findServiceBy(service);
+                        // the service is no more defined, or the logout type is not defined or is back channel
+                        if (registeredService == null || registeredService.getLogoutType() == null
+                                || registeredService.getLogoutType() == LogoutType.BACK_CHANNEL) {
+                            // perform back channel logout
+                            performTypeDependentBackChannelLogout(singleLogoutService, logoutRequest, registeredService);
+                        }
+                    }
+                }
+            }
+        }
+
+        return logoutRequests;
+    }
+
+    void performTypeDependentBackChannelLogout(SingleLogoutService singleLogoutService, LogoutRequest logoutRequest, RegisteredService registeredService) {
+        boolean isLogoutUriEnabledService = registeredService instanceof LogoutUriEnabledRegexRegisteredService;
+        LogoutUriEnabledRegexRegisteredService secondParam;
+        if (isLogoutUriEnabledService) {
+            secondParam = ((LogoutUriEnabledRegexRegisteredService) registeredService);
+        } else {
+            secondParam = null;
+        }
+        boolean successfulLogout = performBackChannelLogout(logoutRequest, secondParam);
+        final LogoutRequestStatus logoutRequestStatus;
+        if (successfulLogout) {
+            logoutRequestStatus = LogoutRequestStatus.SUCCESS;
+        } else {
+            logoutRequestStatus = LogoutRequestStatus.FAILURE;
+            LOGGER.warn("Logout message not sent to [{}]; Continuing processing...",
+                    singleLogoutService.getId());
+        }
+        logoutRequest.setStatus(logoutRequestStatus);
+    }
+
+
+    /**
+     * Log out of a service through back channel.
+     *
+     * @param request the logout request.
+     * @return if the logout has been performed.
+     */
+    boolean performBackChannelLogout(final LogoutRequest request, LogoutUriEnabledRegexRegisteredService registeredService) {
+        final String logoutRequest = this.logoutMessageBuilder.create(request);
+        request.getService().setLoggedOutAlready(true);
+        String originalUrl = request.getService().getOriginalUrl();
```

Comment:
Does the originalUrl contain always the url in a format like https://fqdn/dogu? In development mode, too?

Reply:
The service specific logout URI feature is not supported in development stage yet. See https://github.com/cloudogu/cas/issues/29